crates/eframe/src/native/run.rs

```diff
@@ -270,14 +271,59 @@ fn run_and_return(
     returned_result
 }
 
-fn run_and_exit(event_loop: EventLoop<UserEvent>, mut winit_app: impl WinitApp + 'static) -> ! {
-    log::debug!("Entering the winit event loop (run)…");
+/// An eframe app detached from the event loop.
+///
+/// This is useful to run `eframe` apps while still having control over the event loop.
+/// For example, when you need to have a window managed by `egui` and another managed
+/// by yourself.
+///
+/// The `winit` window is managed internally, and can be accessed through [`Detached::window`].
+/// It can be closed by dropping this object.
+///
+/// The app state and window drawing will only get updated on calls to [`Detached::on_event`].
+pub trait Detached {
+    /// Returns the managed window object
+    fn window(&self) -> Option<&winit::window::Window>;
+    /// Should be called with event loop events.
+    ///
+    /// Events targeted at windows not managed by the app will be ignored.
+    fn on_event(
+        &mut self,
+        event: &Event<'_, UserEvent>,
+        event_loop: &EventLoopWindowTarget<UserEvent>,
+    ) -> Result<DetachedResult>;
+}
 
-    let mut next_repaint_time = Instant::now();
+/// Indicates what is the current state of the app after a [`Detached::on_event`] call.
+pub enum DetachedResult {
+    /// Indicates the next call to [`Detached::on_event`] will result in either a state or paint update.
+    /// Ideally, the event loop control should be set to [`winit::event_loop::ControlFlow::Poll`].
+    UpdateNext,
+    /// Indicates a call to [`Detached::on_event`] will only be required after the specified time.
+    /// Ideally, the event loop control should be set to [`winit::event_loop::ControlFlow::WaitUntil`].
+    UpdateAt(Instant),
+    /// Indicates the app received a close request.
+    Exit,
+}
```

Comment:
Couldn't think of a good way to write these docs, feel free to suggest better ways.

Reply:
Also, this mostly mimics `EventResult` but without `Wait` and `RepaintNow`, since that only makes sense internally. We could easily export that instead and just mention on the docs that those states should be ignored.

Reply:
I think this looks good!